Modules/_json.c

```diff
@@ -314,23 +335,23 @@ escape_unicode(PyObject *pystr)
 }
 
 static void
-raise_errmsg(const char *msg, PyObject *s, Py_ssize_t end)
+raise_errmsg(const char *msg, PyObject *s, Py_ssize_t end, _jsonmodulestate *state)
 {
     /* Use JSONDecodeError exception to raise a nice looking ValueError subclass */
-    static PyObject *JSONDecodeError = NULL;
-    PyObject *exc;
-    if (JSONDecodeError == NULL) {
+    if (state->JSONDecodeError == NULL) {
```

Comment:
Why not simply always getting the attribute? Just remove static and always import the module?

Reply:
IIRC this slows down the call significantly, i'll do some tests to check.

Reply:
Performance version: 1.0.1
Python version: 3.10.0a3+ (32-bit) revision a17f4c7e5a
Report on Windows-10.

Using static JSONDecodeError, vs non static JSONDecodeError:
```
json_dumps: Mean +- std dev: [json-pyid] 21.3 ms +- 2.6 ms -> [json-pyid-noerr] 22.2 ms +- 1.1 ms: 1.04x slower (+4%)
json_loads: Mean +- std dev: [json-pyid] 49.4 us +- 3.4 us -> [json-pyid-noerr] 52.4 us +- 2.0 us: 1.06x slower (+6%)
```
I wonder if the 6% slowdown is worth it?

Reply:
I decided to always import the module, since this should *only* slow down raising errors (which shouldn't happen all the time) :).